subsys/net/lib/http/http_server_http1.c

```diff
@@ -256,6 +257,128 @@ static int dynamic_post_req(struct http_resource_detail_dynamic *dynamic_detail,
 	return 0;
 }
 
+#if defined(CONFIG_FILE_SYSTEM)
+int handle_http1_static_fs_resource(struct http_resource_detail_static_fs *static_fs_detail,
+				    struct http_client_ctx *client)
+{
+#define RESPONSE_TEMPLATE_STATIC_FS                                                                \
+	"HTTP/1.1 200 OK\r\n"                                                                      \
+	"Content-Type: %s%s\r\n\r\n"
+#define CONTENT_ENCODING_GZIP "\r\nContent-Encoding: gzip"
+
+	bool gzipped = false;
+	int len;
+	int remaining;
+	int ret;
+	struct fs_dirent dirent;
+	struct fs_file_t file;
+	char fname[256];
+	char *content_type;
+	/* Add couple of bytes to response template size to have space
+	 * for the content type (15) and encoding
+	 */
+	char http_response[sizeof(RESPONSE_TEMPLATE_STATIC_FS) + 15 +
+			   sizeof(CONTENT_ENCODING_GZIP)];
+	char tmp[TEMP_BUF_LEN];
+
+	if (!(static_fs_detail->common.bitmask_of_supported_http_methods & BIT(HTTP_GET))) {
+		static const char not_found_response[] = "HTTP/1.1 405 Method Not Allowed\r\n"
+							 "Content-Length: 18\r\n\r\n"
+							 "Method Not Allowed";
+
+		ret = http_server_sendall(client, not_found_response,
+					  sizeof(not_found_response) - 1);
+		if (ret < 0) {
+			LOG_DBG("Cannot write to socket (%d)", ret);
+		}
+		return ret;
+	}
+
+	/* get filename amd content-type from url */
+	len = strlen(client->url_buffer);
+	if (len > 1) {
+		if (strcmp(&client->url_buffer[len - 5], ".html") == 0) {
+			content_type = "text/html";
+		} else if (strcmp(&client->url_buffer[len - 3], ".js") == 0) {
+			content_type = "text/javascript";
+		} else if (strcmp(&client->url_buffer[len - 4], ".css") == 0) {
+			content_type = "text/css";
+		} else if (strcmp(&client->url_buffer[len - 4], ".jpg") == 0) {
+			content_type = "image/jpeg";
+		} else if (strcmp(&client->url_buffer[len - 4], ".png") == 0) {
+			content_type = "image/png";
+		} else if (strcmp(&client->url_buffer[len - 4], ".svg") == 0) {
+			content_type = "image/svg+xml";
```

Comment:
Instead of this, we could have an static const array of extensions and content_types, and we could just go through them in a helper function which would be called here. Then adding new content types would be just add new entries to the said array.

Reply:
I have placed the content-types in an iterable section so that users can add new content types themselves.